### PR TITLE
feat: Profile audit logs

### PR DESCRIPTION
**Pull request by ivasilov**
This PR adds audit logs to each user where the user can see history of their actions.

### CODE REVIEW

File: studio/components/interfaces/Account/AuditLogs.tsx [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+  const { data: projects } = useProjectsQuery()
+  const { data: organizations } = useOrganizationsQuery()
+  const { data, error, isLoading, isSuccess, isError, isRefetching, refetch } =
+    useProfileAuditLogsQuery({
+      iso_timestamp_start: dateRange.from,
+      iso_timestamp_end: dateRange.to,
+    })
+
+  // This feature depends on the subscription tier of the user. Free user can view logs up to 1 day
```
Comment (ivasilov) [inline studio/components/interfaces/Account/AuditLogs.tsx:39]: If this looks like a good solution, we should also copy it over to the organization level logs screen.

File: studio/data/organizations/organization-audit-logs-query.ts [hunk 2]
```diff
@@ -32,7 +33,7 @@ export type OrganizationAuditLog = {
 }
 
 export type OrganizationAuditLogsResponse = {
-  result: OrganizationAuditLog[]
+  result: AuditLog[]
   retention_period: number
 }
```
Comment (ivasilov) [inline studio/data/organizations/organization-audit-logs-query.ts:39]: `iso_timestamp_start` and `iso_timestamp_end` are not actually optional. The API sends an error if you don't supply them.

File: studio/data/profile/profile-audit-logs-query.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  retention_period: number
+}
+
+export const useProfileAuditLogsQuery = <TData = ProfileAuditLogsData>(
+  vars: ProfileAuditLogsVariables,
+  options: UseQueryOptions<ProfileAuditLogsData, ProfileAuditLogsError, TData> = {}
+) => {
+  const { iso_timestamp_start, iso_timestamp_end } = vars
+  return useQuery<ProfileAuditLogsData, ProfileAuditLogsError, TData>(
+    profileKeys.auditLogs({ date_start: iso_timestamp_start, date_end: iso_timestamp_end }),
```
Comment (ivasilov) [inline studio/data/profile/profile-audit-logs-query.ts:44]: The keys should include the date in full (including the time).

File: studio/components/interfaces/Account/AuditLogs.tsx [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+import Table from 'components/to-be-cleaned/Table'
+import AlertError from 'components/ui/AlertError'
+import { DatePicker } from 'components/ui/DatePicker'
+import ShimmeringLoader from 'components/ui/ShimmeringLoader'
+import { AuditLog } from 'data/organizations/organization-audit-logs-query'
+import { useOrganizationsQuery } from 'data/organizations/organizations-query'
+import { useProfileAuditLogsQuery } from 'data/profile/profile-audit-logs-query'
+import { useProjectsQuery } from 'data/projects/projects-query'
+import { Alert, Button, IconArrowDown, IconArrowUp, IconRefreshCw } from 'ui'
+import FilterPopover from '../Organization/AuditLogs/FilterPopover'
```
Comment (ivasilov) [inline studio/components/interfaces/Account/AuditLogs.tsx:15]: Reusing these files from `Organization` folder. Is there a better place for them?

File: studio/components/interfaces/Account/AuditLogs.tsx (hunk 1 above)
Comment (joshenlim) [reply studio/components/interfaces/Account/AuditLogs.tsx:39]: yup i think this is good actually 👍

File: studio/components/interfaces/Account/AuditLogs.tsx (hunk 4 above)
Comment (joshenlim) [reply studio/components/interfaces/Account/AuditLogs.tsx]: yeap - shifted them to interfaces/AuditLogs

File: studio/data/organizations/organization-audit-logs-query.ts [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
 import { get } from 'lib/common/fetch'
 import { API_URL } from 'lib/constants'
-import { useCallback } from 'react'
-import { organizationKeys } from './keys'
 import { ResponseError } from 'types'
-import dayjs from 'dayjs'
+import { organizationKeys } from './keys'
 
-export type OrganizationAuditLog = {
+export type AuditLog = {
```
[the remark this answers is not in the document]
Comment (joshenlim) [reply studio/data/organizations/organization-audit-logs-query.ts:10]: yeah once we move to data/fetchers should be fixed

**Comment by joshenlim** [comment]
@ivasilov generally lgtm! there's one very very odd UI bug though that I'm not too sure why its happening
so if you open the org audit logs, the styling for the DatePicker button seems to be lost, and likewise if you navigate to the account audit logs from there

image

however, if you refresh the page while on the account audit logs, the styles are there just fine
image

**Comment by joshenlim** [comment]
okay no worries - managed to fix the issue

**Review by joshenlim — APPROVED** [review]

**Review by joshenlim — APPROVED** [review]

**Pull request merged**